src/libstd/ffi/c_str.rs

```diff
@@ -314,9 +314,11 @@ impl CString {
 }
 
 // Turns this `CString` into an empty string to prevent
-// memory unsafe code from working by accident.
+// memory unsafe code from working by accident. Inline
+// to prevent LLVM from optimizing it away in debug builds.
 #[stable(feature = "cstring_drop", since = "1.13.0")]
 impl Drop for CString {
+    #[inline(always)]
```

Comment:
I think `#[inline]` should be plenty here in general (`always` is best left to LLVM to decide)